### PR TITLE
feat: added the contribution guidelines

### PR DESCRIPTION
**Pull request by palashgdev**
Close https://github.com/SigNoz/signoz/issues/2849

### CODE REVIEW

File: frontend/CONTRIBUTIONS.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+10. memoize functions, and variables judiciously. Over-optimizations can also lead to performance issues.
+
+11. Try to avoid inline functions as much as possible. It will increase the readability and thus more comprehensible.
+
+12. Avoid using multiple ifs' or cases in the switch. Try to create a mapper out of it and segregate in multiple functions.
+
+13. if function expects more than 3 arguments wrap them in an object(except when it is memoized).
+
+14. use **useMemo** and **useCallback** to memorise accordingly
```
Comment (GermaVinsmoke) [inline frontend/CONTRIBUTIONS.md:29]: memoise*

**Comment by GermaVinsmoke** [comment]
Adding code snippets of **What to do** and **What not to do** for all the points can help in understanding what we want the contributors to do and what not to do.

For example - 
- API/Service files should not have business logic. It removes the flexibility for the consumer to handle it differently.

The thought process is still not clear, so maybe with the help of an API example, it can show how to do it. 

- Try to internationalize the strings

A ready to use code snippet example can help in iterating the things quickly.

**Comment by GermaVinsmoke** [comment]
I think dividing the whole thing into different sections can be beneficial.

### Styling
- Avoid inline styling
- Use rem instead of px
- ...
### React and Components
- Make component as small as possible (and divide those into individual one)
- memoize functions, and variables judiciously. Over-optimizations can also lead to performance issues.
- ...

### API and Services
- API/Service files should not have business logic. It removes the flexibility for the consumer to handle it differently.
- use useQuery for fetching data and useMutation for updating data
- ...

### Linting and setup
- Never disable eslint and typescript error (if we are disabling any rule need to write why we are disabling any rule)
- ...

### Miscellaneous
- No function should exceed more than 40 lines
- try to internationalize all strings
- ...

**Comment by srikanthccv** [comment]
I agree with the last point, i.e dividing this into different sections is good.

**Comment by palashgdev** [comment]
@srikanthccv @GermaVinsmoke this can be addressed in separate issue

**Comment by GermaVinsmoke** [comment]
> @srikanthccv @GermaVinsmoke this can be addressed in separate issue

Alright. I can do this thing when it gets merged.

**Comment by srikanthccv** [comment]
I mean, this is not like a feature or something to address in multiple commits. Why not just address it in the same PR? I will leave it to you folks.

**Comment by palashgdev** [comment]
@> > @srikanthccv @GermaVinsmoke this can be addressed in separate issue
> 
> 
> 
> Alright. I can do this thing when it gets merged.

@GermaVinsmoke can you raise the pr to this branch?

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**